datafusion/core/tests/sqllogictests/test_files/array.slt

```diff
@@ -204,3 +204,21 @@ query II rowsort
 select array_ndims(array_fill(1, [1, 2, 3])), array_ndims([[[[[[[[[[[[[[[[[[[[[1]]]]]]]]]]]]]]]]]]]]]);
 ----
 3 21
+
+# array concatenate operator #1 (like array_concat scalar function)
```

Comment:
I recommend adding a test for `null handling in these arrays 

like 

```sql
select make_array(1, 2, 3) || make_array(4, null, 6);
```

and

```sql
select make_array(1, 2, 3) || null;
```